src/transport.ts

```diff
@@ -88,71 +89,73 @@ export class PinoSentryTransport {
     });
   }
 
-  public prepareAndGo(chunk: any, cb: any): void {
-    const severity = this.getLogSeverity(chunk.level);
+  public prepareAndGo(chunkInfo: ChunkInfo, cb: any): void {
+    chunkInfo.run((chunk) => {
+      const severity = this.getLogSeverity(chunk.level);
 
-    // Check if we send this Severity to Sentry
-    if (this.shouldLog(severity) === false) {
-      setImmediate(cb);
-      return;
-    }
-
-    const tags = chunk.tags || {};
-    const breadcrumbs: Breadcrumb[] = chunk.breadcrumbs || {};
-
-    if (chunk.reqId) {
-      tags.uuid = chunk.reqId;
-    }
-
-    if (chunk.responseTime) {
-      tags.responseTime = chunk.responseTime;
-    }
-
-    if (chunk.hostname) {
-      tags.hostname = chunk.hostname;
-    }
-
-    const extra: any = {};
-    this.extraAttributeKeys.forEach((key: string) => {
-      if(chunk[key] !== undefined) {
-        extra[key] = chunk[key];
+      // Check if we send this Severity to Sentry
+      if (this.shouldLog(severity) === false) {
+        setImmediate(cb);
+        return;
       }
-    });
-    const message = get(chunk, this.messageAttributeKey);
-    const stack = get(chunk, this.stackAttributeKey) || '';
 
-    const scope = new Sentry.Scope();
-    this.decorateScope(chunk, scope);
-
-    scope.setLevel(severity);
-
-    if (this.isObject(tags)) {
-      Object.keys(tags).forEach(tag => scope.setTag(tag, tags[tag]));
-    }
+      const tags = chunk.tags || {};
+      const breadcrumbs: Breadcrumb[] = chunk.breadcrumbs || {};
 
-    if (this.isObject(extra)) {
-      Object.keys(extra).forEach(ext => scope.setExtra(ext, extra[ext]));
-    }
+      if (chunk.reqId) {
+        tags.uuid = chunk.reqId;
+      }
 
-    if (this.isObject(breadcrumbs)) {
-      Object.values(breadcrumbs).forEach(breadcrumb => scope.addBreadcrumb(breadcrumb));
-    }
+      if (chunk.responseTime) {
+        tags.responseTime = chunk.responseTime;
+      }
 
-    // Capturing Errors / Exceptions
-    if (this.isSentryException(severity)) {
-      const error = message instanceof Error ? message : new ExtendedError({ message, stack });
+      if (chunk.hostname) {
+        tags.hostname = chunk.hostname;
+      }
 
-      setImmediate(() => {
-        Sentry.captureException(error, scope);
-        cb();
-      });
-    } else {
-      // Capturing Messages
-      setImmediate(() => {
-        Sentry.captureMessage(message, scope);
-        cb();
+      const extra: any = {};
+      this.extraAttributeKeys.forEach((key: string) => {
+        if(chunk[key] !== undefined) {
+          extra[key] = chunk[key];
+        }
```

Comment:
Shouldn't the `forEach` close after this?

Reply:
Yep :grimace:

Reply:
@segevfiner cheers! By the way, we're currently using this library to add additional contextual data to sentry errors (from the logger). As it stands, none of the data is being sent to sentry. Is this the actual problem this PR is addressing?

Reply:
@jesushernandez you should create issue of your problem with steps to reproduce to make the problem visible.

Reply:
I'll try to reproduce today and get back to you!